src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,73 @@
+package duke;
+import duke.command.DukeCommands;
+import duke.util.*;
+
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class Duke {
+    /**
+     * Simple string array to store inputs
+     **/
+    private final DukeTaskList list;
+
+    private final DukeDB database;
+
+    private final DukeConfig config;
+
+    private final Ui ui;
+    /**
+     * Basic constructor to initialise the list
+     **/
+    public Duke(DukeDB database, DukeConfig config) {
+        this.ui = new Ui();
+        this.database = database;
+        this.list = new DukeTaskList(database.load().orElse(new ArrayList<>()));
+        this.config = config;
+
+    }
+
+
+    public Duke(DukeDB database) {
+        this.ui = new Ui();
+        this.database = database;
+        this.list = new DukeTaskList(database.load().orElse(new ArrayList<>()));
+        this.config = new DukeConfig();
+    }
+
+
+    /**
+     * Simple formatting tool to be used when printing commands
+     *
+     * @param text The text to be printed.
+     */
+    public static void printMsg(String text) {
+        String bar = "===============================================";
+        System.out.printf("%s\n%s\n%s\n", bar, text, bar);
+    }
+
+    /**
+     * Method that listens and scans for user input.
+     * Program will exit when the command "bye" is given to the scanner.
+     */
+    public void listen(){
+        Scanner scanner = new Scanner(System.in);
+        this.ui.greet(this.list.getSize());
```

Comment:
Could consider omitting the `this` keyword here? My tutor mentioned to avoid unnecessary usage of `this`, but I would personally not mind using `this` as well unless it gets too cluttered.

Reply:
hmmm, I would think `this` helps to keep things cleaner as it can be easily recognized as an object instance variable as compared to the other variables.

but I do agree that it might be unnecessary as it doesn't serve any purpose. 
however, I will keep `this` for now